### PR TITLE
feature/CP-6134-app-banners-unsubscribe-trigger-android

### PR DESCRIPTION
**Pull request by unnaticleverpush**
App banner add unsubscribe trigger

### CODE REVIEW

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]
does this work if subscriber removes notification permission?

**Comment by unnaticleverpush** [comment]
Yes. The Banner will still appear even if notification permission is not allowed.

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/responsehandlers/UnsubscribeResponseHandler.java [hunk 1]
```diff
@@ -6,6 +6,9 @@
 import com.cleverpush.CleverPush;
 import com.cleverpush.CleverPushHttpClient;
 import com.cleverpush.CleverPushPreferences;
+import com.cleverpush.Constants;
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/responsehandlers/UnsubscribeResponseHandler.java:9]: These imports are used?

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 2]
```diff
@@ -1361,6 +1361,8 @@ public void unsubscribe(UnsubscribedListener listener) {
 
       CleverPushHttpClient.post("/subscription/unsubscribe", jsonBody,
           new UnsubscribeResponseHandler(this, listener).getResponseHandler());
+    } else {
+      clearSubscriptionData();
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:1365]: Can you explain this change?

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 2 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1365]: I am doing the same as the IOS code for unsubscribe

https://github.com/cleverpush/cleverpush-ios-sdk/blob/404a5aa405aedca7b29be83d27c62f3449032e1b/CleverPush/Source/CleverPushInstance.m#L1032

File: cleverpush/src/main/java/com/cleverpush/responsehandlers/UnsubscribeResponseHandler.java (hunk 1 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/responsehandlers/UnsubscribeResponseHandler.java:9]: No. I'll remove it.

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 2 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1365]: @gbhrdt Should I remove this?

**Pull request merged**